### PR TITLE
set of patch fixing gromacs build issues when cp2k is build with spack and cmake

### PR DESCRIPTION
**Pull request by mtaillefumier**
@haampie and I have been working on this set of patches that fix various issues when cp2k is build with cmake and used as dependency in gromacs. 

- libcp2k.pc has now all the right information
- fix various typo in cp2kConfig.cmake (when we use `find_package(cp2k)`)
-  fix the C and C++ std as well. It should one issue in the spack recipe
- fix the c++ standard in the toolchain.
- move the findPackage.cmake to a dedicated directory.
- fix installation directories for modules.
this PR includes PR #3041

Co-authored-by: Harmen Stoppels <me@harmenstoppels.nl>

### CODE REVIEW

File: tools/toolchain/scripts/generate_arch_files.sh [hunk 1]
```diff
@@ -132,7 +132,7 @@ else
   FC_arch+=" IF_MPI(-fc=${I_MPI_FC}|)"
   LD_arch+=" IF_MPI(-fc=${I_MPI_FC}|)"
   CFLAGS="${G_CFLAGS} -std=c11 -Wall \$(DFLAGS)"
-  CXXFLAGS="${G_CFLAGS} -std=c11 -Wall \$(DFLAGS)"
+  CXXFLAGS="${G_CFLAGS} -std=c++11 -Wall \$(DFLAGS)"
```
Comment (RMeli) [inline tools/toolchain/scripts/generate_arch_files.sh]: In CMake you set `set(CMAKE_CXX_STANDARD 14)`.
```suggestion
  CXXFLAGS="${G_CFLAGS} -std=c++14 -Wall \$(DFLAGS)"
```
?

**Comment by mtaillefumier** [comment]
i removed the option to build dbcsr with cp2k algether. The cmake build will fail for sure.

**Comment by haampie** [comment]
Thanks, that seems to work well

**Comment by mtaillefumier** [comment]
except that their CI uses a functionality I just deleted.... we have to switch to spack...

**Comment by oschuett** [comment]
We're actually almost ready to build DBCSR via the toolchain (#351). All that's left is to hook up [this script](https://github.com/cp2k/cp2k/blob/master/tools/toolchain/scripts/stage9/install_dbcsr.sh) to [here](https://github.com/cp2k/cp2k/blob/6b94574d37ca4648fbb0f33b2f35874362a23d3a/tools/toolchain/install_cp2k_toolchain.sh#L979) and [here](https://github.com/cp2k/cp2k/blob/6b94574d37ca4648fbb0f33b2f35874362a23d3a/tools/docker/generate_dockerfiles.py#L774).

**Comment by RMeli** [comment]
@mtaillefumier, since you are already touching CMake, would you mind adding the following?
```diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7126fa42a..a2664e2b5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1585,6 +1585,7 @@ target_compile_definitions(
          $<$<CONFIG:COVERAGE>:__NO_ABORT>
          $<$<CONFIG:DEBUG>:__HAS_IEEE_EXCEPTIONS>
          $<$<CONFIG:DEBUG>:__CHECK_DIAG>
+         $<$<BOOL:${CP2K_USE_MPI_F08}>:__MPI_F08>
          $<$<BOOL:${CP2K_USE_PEXSI}>:__PEXSI>
          $<$<BOOL:${CP2K_USE_PLUMED2}>:__PLUMED2>
          $<$<BOOL:${CP2K_USE_QUIP}>:__QUIP>
```

Otherwise CP2K doesn't compile with the latest version of MPICH (which requires `-DCP2K_USE_MPI_F08=ON`.

Related to https://github.com/cp2k/dbcsr/issues/661.

**Comment by mtaillefumier** [comment]
> We're actually almost ready to build DBCSR via the toolchain (#351). All that's left is to hook up [this script](https://github.com/cp2k/cp2k/blob/master/tools/toolchain/scripts/stage9/install_dbcsr.sh) to [here](https://github.com/cp2k/cp2k/blob/6b94574d37ca4648fbb0f33b2f35874362a23d3a/tools/toolchain/install_cp2k_toolchain.sh#L979) and [here](https://github.com/cp2k/cp2k/blob/6b94574d37ca4648fbb0f33b2f35874362a23d3a/tools/docker/generate_dockerfiles.py#L774).

thanks I will take care of it then.

**Comment by mtaillefumier** [comment]
> ```diff
> $<$<BOOL:${CP2K_USE_MPI_F08}>:__MPI_F08>
> ```

it is already there. But it is `-DCP2K_ENABLE_MPI_F08` not `-DCP2K_USE_MPI_F08`

**Comment by mtaillefumier** [comment]
the toolchain is giving me some resistance. I would propose to ignore the toolchain for the time being and merge this PR after I fix the compilation issue. 
```

 341 |          CALL mp_group%set_handle(eri%mp_group%get_handle())
      |                                                          1
Error: 'get_handle' at (1) is not an inquiry reference to an intrinsic type component 'mp_group'
```

any idea why it fails now ?

**Comment by oschuett** [comment]
> any idea why it fails now ?

It's probably a botched merge with d5cf19bccd0d30412fceb27a31dc6645d5b963f9. Try to rebase your branch onto the latest master.

**Comment by mtaillefumier** [comment]
> > any idea why it fails now ?
> 
> It's probably a botched merge with [d5cf19b](https://github.com/cp2k/cp2k/commit/d5cf19bccd0d30412fceb27a31dc6645d5b963f9). Try to rebase your branch onto the latest master.

Thanks. I just rebased to master. Let's give it a try. I am also trying to fix the compilation warnings with fromtran flags been passed to gcc. I can not reproduce it locally though

**Comment by mtaillefumier** [comment]
I am also passing the compilation flags the proper way instead of tinkering with CMAKE_FORTRAN_FLAGS.... Sorry if it is taking so long.

**Comment by mtaillefumier** [comment]
I found the root of the problem. DBCSR should be 2.6.0 not 2.5.0

**Comment by mtaillefumier** [comment]
I will have to continue that tomorrow

**Comment by mtaillefumier** [comment]
thanks

**Pull request merged**